src/main/java/org/jenkinsci/plugins/workflow/support/storage/FlowNodeStorage.java

```diff
@@ -28,26 +28,87 @@
 import org.jenkinsci.plugins.workflow.flow.FlowExecution;
 import org.jenkinsci.plugins.workflow.graph.FlowActionStorage;
 import org.jenkinsci.plugins.workflow.graph.FlowNode;
+import org.jenkinsci.plugins.workflow.steps.StepDescriptor;
+
 import java.io.IOException;
 import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
 
 /**
  * Abstraction of various ways to persist {@link FlowNode}, for those {@link FlowExecution}s
- * who wants to store them within Jenkins.
- *
- * A flow graph has a characteristic that it's additive.
+ * who want to store them within Jenkins.
  *
- * This is clearly a useful internal abstraction to decouple {@link FlowDefinition} implementation
- * from storage mechanism, but not sure if this should be exposed to users.
+ * A flow graph has a characteristic that it is additive.
  *
+ * Flow nodes may be stored in memory or directly persisted to disk at any given moment, but invoking {@link #flush()}
+ *  should always guarantee that everything currently in memory is written.
  * @author Kohsuke Kawaguchi
+ * @author Sam Van Oort
  */
 public abstract class FlowNodeStorage implements FlowActionStorage {
+    // Set up as "avoid" because an unset field will default to false.
```

Comment:
Seems to contradict the actual value?

Reply:
@abayer Not sure what you mean, but clarifying the  comment, thanks

Reply:
The impression I got from reading the comment was that you'd be setting the field as true here, since if the default value for an unset field is false, why bother mentioning it when you're explicitly setting it to false anyway? Also, since you're setting it to false, you're setting it up as "don't avoid". =)